src/roswire/ros2/node_manager.py

```diff
@@ -1,30 +1,38 @@
 # -*- coding: utf-8 -*-
 __all__ = ('ROS2NodeManager',)
 
-from typing import Iterator, Mapping
+from typing import Iterator
 import typing
 
 from loguru import logger
 import attr
 
 from .node import ROS2Node
+from .state import ROS2StateProbe
+from ..proxy import SystemState
 from .. import exceptions as exc
+from ..interface import NodeManager
 
 if typing.TYPE_CHECKING:
     from ..app import AppInstance
 
 
 @attr.s(frozen=True, auto_attribs=True, slots=True)
-class ROS2NodeManager(Mapping[str, ROS2Node]):
+class ROS2NodeManager(NodeManager):
     """Provides an interface for interacting with ROS2 nodes."""
-    app_instance: 'AppInstance'
+    app_instance: 'AppInstance' = attr.ib()
+    _state_probe: 'ROS2StateProbe' = attr.ib(init=False)
```

Comment:
Adding a comment here just to make sure that we remember to include an `__attrs_post_init__` method before merging.